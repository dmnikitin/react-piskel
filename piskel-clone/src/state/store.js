import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { getFromLocalStorage } from '../helpers/localStorageHandler';
import combined from './reducers/combined';

const localStorageData = getFromLocalStorage();
const store = createStore(combined, localStorageData, composeWithDevTools());

export default store;
