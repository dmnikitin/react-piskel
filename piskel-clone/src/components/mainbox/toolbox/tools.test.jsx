import React from 'react'
import { shallow, mount } from 'enzyme';
import Tools from './tools';
import store from '../../../state/store';

const callback = () => { };
jest.mock('../button.jsx', () => 'Button');
const wrapperShallow = shallow(<Tools callback={callback} />);
const wrapperMount = mount(<Tools store={store} callback={callback} />);

describe('tools component', () => {
  it('renders correctly', () => {
    expect(wrapperShallow).toMatchSnapshot();
  });
  it("displays initial div", () => {
    expect(wrapperMount.find("div")).toHaveLength(1);
  });
  it('displays proper amount of buttons', () => {
    expect(wrapperMount.find("div").children()).toHaveLength(6);
  });
});
