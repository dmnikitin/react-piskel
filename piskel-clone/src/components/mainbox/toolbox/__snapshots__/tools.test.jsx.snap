// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tools component renders correctly 1`] = `
<div
  className="toolbox-tools"
>
  <Button
    callback={[Function]}
    data="pen"
    icon="pencil"
  />
  <Button
    callback={[Function]}
    data="eraser"
    icon="eraser"
  />
  <Button
    callback={[Function]}
    data="colorPicker"
    icon="eyedropper"
  />
  <Button
    callback={[Function]}
    data="stroke"
    icon="arrows-h"
  />
  <Button
    callback={[Function]}
    data="bucket"
    icon="shower"
  />
  <Button
    callback={[Function]}
    data="allToOneColor"
    icon="magic"
  />
</div>
`;
